Code/Framework/AzCore/AzCore/IO/Streamer/RequestPath.cpp

```diff
@@ -121,19 +108,19 @@ namespace AZ
                 AZ_Assert(FileIOBase::GetInstance(),
                     "Trying to resolve a path in RequestPath before the low level file system has been initialized.");
 
-                char fullPath[AZ::IO::MaxPathLength];
-                if (!FileIOBase::GetInstance()->ResolvePath(m_path.c_str(), fullPath, AZ_ARRAY_SIZE(fullPath)))
+                AZStd::optional<AZ::IO::FixedMaxPath> fullPath = FileIOBase::GetInstance()->ResolvePath(m_path);
+                if (!fullPath)
                 {
                     m_absolutePathHash = s_invalidPathHash;
                     return;
                 }
 
-                size_t relativePathLength = m_path.length() - m_relativePathOffset;
-                m_path = fullPath;
-                m_absolutePathHash = AZStd::hash<AZStd::string>{}(m_path);
-                if (m_path.length() >= relativePathLength)
+                size_t relativePathLength = m_path.Native().length() - m_relativePathOffset;
+                m_path = fullPath.value();
+                m_absolutePathHash = AZStd::hash<decltype(m_path)>{}(m_path);
```

Comment:
You can avoid the decltype by using the [hash_value](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L650-L655) function
```suggestion
                m_absolutePathHash = AZ::IO::hash_value(m_path);
```